Controllers/AdminController.cs

```diff
@@ -33,13 +33,22 @@ public ActionResult Search(SearchVM searchVM)
                 // TODO: Order the results by Title
             }
 
+
             //TODO: Add similar logic to filter foundbooks collection by last part of the Author's Name, if given
             // (HINT: consider the EndsWith() method, also adjust the Search View and ViewModel to add items)
-
+            if (searchVM.Author!= null)
+            {
+                foundBooks = foundBooks
+                            .Where(b => b.Author.Contains(searchVM.Author));
+            }
             //TODO: Filter the collection by price between a low and high value, if given
             //       order results by descending price 
             // (Note: you will need to adjust the Search ViewModel and View to add search fields)
-
+            if (searchVM.MinPrice!= null)
+            {
+                foundBooks = foundBooks
+                            .Where(b )
```

Comment:
For the logic, you can look [online for the logical operator syntax](https://www.tutlane.com/tutorial/csharp/csharp-logical-operators-with-examples), to look for  what comes after the 
`foundBooks.Where(b => ... );